### ISSUE TITLE
Transport Variables can not be modified for invoice and shipment mail

### ISSUE THREAD

**Issue by staywithabhi** [magento/magento2#15152]
I am using magento 2.2.3 and have custom shipping method to magento store,every thing is working like a charm except custom details in invoice and shipment mail.Transport variables are modified for order email but not working for invoice and shipment.I have crosschecked and observer is working fine for invoice mail and shipment mail.
### Preconditions

1. 
2. 

### Steps to reproduce

1. 
2. 
3. 

### Expected result

1. Transport variables must be modified for invoice and shipment mail and custom details must be sent in invoice and shipment mail.

### Actual result

1. [Screenshot, logs]
Transport variables not modified and custom data not sent in mail

**Comment by gwharton** [comment]
Setting of the transport variables was broken for ages.

An attempt to fix it was made however the fix did not cover all aspects properly. The failed fix was released in 2.2.4.

Here's the original bug report. https://github.com/magento/magento2/issues/10210
Here is a link to the PR that partially fixed the problem https://github.com/magento/magento2/pull/12132
Here is a link to the pull request that fixes the rest of the problem https://github.com/magento/magento2/pull/15040

That last PR is waiting to be merged, but hopefully as of 2.2.5 all email senders will properly work with the new transportObject data parameter.

Note that following this change, the use of $transport is deprecated. You should use $transportObject instead in your observer.

Here is an example of a working observer that adds plain text billing and shipping address to the transport object.

```
    public function __construct(
      \Magento\Sales\Model\Order\Address\Renderer $addressRenderer,
      \Magento\Payment\Helper\Data $paymentHelper,
      \Magento\Sales\Model\Order\Email\Container\OrderIdentity $identityContainer
    ) {
            $this->addressRenderer = $addressRenderer;
            $this->paymentHelper = $paymentHelper;
            $this->identityContainer = $identityContainer;
    }

    public function execute(\Magento\Framework\Event\Observer $observer)
    {
        $transportObject = $observer->getEvent()->getData('transportObject');
        $order = $transportObject->getData('order');
        $shippingaddress = $order->getIsVirtual() ? null : $this->addressRenderer->format($order->getShippingAddress(), 'text');
        $billingaddress = $this->addressRenderer->format($order->getBillingAddress(), 'text');        
        $shippingaddress = preg_replace("/(^[\r\n]*|[\r\n]+)[\s\t]*[\r\n]+/", "\n", $shippingaddress);
        $billingaddress = preg_replace("/(^[\r\n]*|[\r\n]+)[\s\t]*[\r\n]+/", "\n", $billingaddress);
        $transportObject->setData('textShippingAddress', $shippingaddress);
        $transportObject->setData('textBillingAddress', $billingaddress);        
    }

```

**Comment by staywithabhi** [comment]
Hi gwharton ,
Thanks for the help and references provided.Issue has been fixed by using **transportObject**.
Hope to see changes in latest magento release
Thank You.
